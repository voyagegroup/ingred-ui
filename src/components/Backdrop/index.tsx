export { default, BackdropProps } from "./Backdrop";
