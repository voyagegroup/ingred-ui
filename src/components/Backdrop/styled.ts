import styled, { css } from "styled-components";

export const Container = styled.div<{ invisible: boolean }>`
  position: fixed;
  display: flex;
  align-items: center;
  justify-content: center;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  /* TODO: color */
  background-color: #001326cb;

  ${({ invisible }) =>
    invisible &&
    css`
      background-color: transparent;
    `}
`;
