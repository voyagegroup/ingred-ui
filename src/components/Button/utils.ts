import { Theme } from "../../themes";
import { ButtonColor, ButtonColorStyle } from "./Button";

export const getContainerColorStyles = (
  theme: Theme,
): { [P in ButtonColor]: ButtonColorStyle } => ({
  primary: {
    normal: {
      background: theme.palette.primary.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowPrimary"],
      border: `1px solid ${theme.palette.primary.dark}`,
    },
    hover: {
      background: theme.palette.primary.dark,
      border: "none",
    },
    active: {
      background: theme.palette.primary.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  basicLight: {
    normal: {
      background: theme.palette.background.default,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.divider}`,
    },
    hover: {
      background: theme.palette.gray.highlight,
      border: `1px solid ${theme.palette.divider}`,
    },
    active: {
      background: theme.palette.gray.highlight,
      boxShadow: theme.shadow["3dShadowActive"],
      border: `1px solid ${theme.palette.divider}`,
    },
  },
  basicDark: {
    normal: {
      background: theme.palette.basicDark.main,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.basicDark.deepDark}`,
    },
    hover: {
      background: theme.palette.basicDark.dark,
      border: "none",
    },
    active: {
      background: theme.palette.basicDark.dark,
      boxShadow: theme.shadow["3dShadowBasicActive"],
      border: "none",
    },
  },
  danger: {
    normal: {
      background: theme.palette.danger.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowDanger"],
      border: `1px solid ${theme.palette.danger.dark}`,
    },
    hover: {
      background: theme.palette.danger.dark,
      border: "none",
    },
    active: {
      background: theme.palette.danger.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  clear: {
    normal: {
      background: "none",
      color: theme.palette.black,
      boxShadow: "none",
      border: "none",
    },
    hover: {
      background: theme.palette.gray.light,
      border: "none",
    },
    active: {
      background: theme.palette.gray.main,
      boxShadow: "none",
      border: "none",
    },
  },
  primaryPale: {
    normal: {
      background: theme.palette.primaryPale.main,
      color: theme.palette.primaryPale.dark,
      boxShadow: theme.shadow["3dShadowPrimaryPale"],
      border: `1px solid ${theme.palette.primaryPale.light}`,
    },
    hover: {
      background: theme.palette.primaryPale.highlight,
      border: "none",
    },
    active: {
      background: theme.palette.primaryPale.highlight,
      boxShadow: theme.shadow["3dShadowPrimaryPaleActive"],
      border: "none",
    },
  },
});
