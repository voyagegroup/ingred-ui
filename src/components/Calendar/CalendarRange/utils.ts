import dayjs, { Dayjs } from "dayjs";

const isStart = (startDate: Dayjs | null, month: Dayjs, day: number) =>
  startDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isEnd = (endDate: Dayjs | null, month: Dayjs, day: number) =>
  endDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isBetween = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) =>
  (startDate &&
    endDate &&
    dayjs(new Date(month.year(), month.month(), day)).isAfter(
      startDate.format("YYYY-MM-DD"),
      "day",
    ) &&
    dayjs(new Date(month.year(), month.month(), day)).isBefore(
      endDate.format("YYYY-MM-DD"),
      "day",
    )) ??
  false;

export const getDayState = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) => {
  if (isStart(startDate, month, day)) {
    return "start";
  }
  if (isEnd(endDate, month, day)) {
    return "end";
  }
  if (isBetween(startDate, endDate, month, day)) {
    return "between";
  }
  return "none";
};
