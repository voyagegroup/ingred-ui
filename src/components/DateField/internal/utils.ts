export const getInputWidth = (input: string) =>
  [...input.split("")].reduce((acc, char) => {
    const code = char.charCodeAt(0);
    return acc + (code >= 0x3000 && code <= 0xff60 ? 2 : 1);
  }, 0);
