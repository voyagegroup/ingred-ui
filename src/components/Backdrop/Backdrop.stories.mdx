import { Meta, Story, ArgsTable, Canvas } from "@storybook/addon-docs/blocks";
import Backdrop from "./Backdrop";
import Spinner from "../Spinner";
import Button from "../Button";

<Meta
  title="Components/Feedback/Backdrop"
  component={Backdrop}
  parameters={{ docs: { source: { type: "code" } } }}
/>

# Backdrop

<ArgsTable of={Backdrop} />

## Samples

<Canvas>
  <Story
    name="Example"
    args={{
      isOpen: false,
    }}
  >
    {(args) => {
      const [isOpen, setIsOpen] = React.useState(false);
      const handleToggle = () => {
        setIsOpen(!isOpen);
      };
      return (
        <>
          <Button onClick={handleToggle}>Toggle Show Backdrop & Spinner</Button>
          <Backdrop
            {...args}
            isOpen={isOpen || args.isOpen}
            onClick={handleToggle}
          >
            <Spinner />
          </Backdrop>
        </>
      );
    }}
  </Story>
</Canvas>
