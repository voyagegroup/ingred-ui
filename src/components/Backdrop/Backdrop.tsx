import * as React from "react";
import * as Styled from "./styled";
import Fade from "../Fade";
import { CSSTransitionProps } from "../../utils/reactTransitionGroup";

export type BackdropProps = React.ComponentPropsWithoutRef<"div"> & {
  isOpen?: boolean;
  invisible?: boolean;
  /**
   * Unit: millisecond
   */
  transitionDuration?: number;
  /**
   * props of [Fade](/?path=/docs/components-utils-fade)
   */
  fadeProps?: CSSTransitionProps;
};

const Backdrop = React.forwardRef<HTMLDivElement, BackdropProps>(
  (
    {
      isOpen = true,
      invisible = false,
      transitionDuration,
      children,
      fadeProps,
      ...rest
    },
    ref,
  ) => {
    return (
      <Fade
        in={isOpen}
        timeout={transitionDuration}
        unmountOnExit={true}
        {...fadeProps}
      >
        <Styled.Container invisible={invisible} {...rest} ref={ref}>
          {children}
        </Styled.Container>
      </Fade>
    );
  },
);

export default Backdrop;
