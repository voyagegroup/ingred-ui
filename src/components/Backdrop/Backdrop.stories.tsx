import React from "react";
import { StoryObj } from "@storybook/react";
import { Title, ArgsTable, Stories } from "@storybook/addon-docs";
import Backdrop, { BackdropProps } from "./Backdrop";
import Spinner from "../Spinner";
import Button from "../Button";

export default {
  title: "Components/Feedback/Backdrop",
  components: Backdrop,
  args: {
    isOpen: false,
  },
  parameters: {
    docs: {
      source: { language: "tsx" },
      page: () => (
        <>
          <Title />
          <ArgsTable of={Backdrop} />
          <Stories includePrimary title="Stories" />
        </>
      ),
    },
  },
};

export const Example: StoryObj<BackdropProps> = {
  render: (args) => {
    const [isOpen, setIsOpen] = React.useState(false);
    const handleToggle = () => {
      setIsOpen(!isOpen);
    };
    return (
      <>
        <Button onClick={handleToggle}>Toggle Show Backdrop & Spinner</Button>
        <Backdrop
          {...args}
          isOpen={isOpen || args.isOpen}
          onClick={handleToggle}
        >
          <Spinner />
        </Backdrop>
      </>
    );
  },
};
