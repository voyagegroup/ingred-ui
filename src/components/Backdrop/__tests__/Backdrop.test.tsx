import * as React from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";
import Backdrop from "../";

describe("Backdrop component testing", () => {
  afterEach(cleanup);

  test("Backdrop", () => {
    const { asFragment } = renderWithThemeProvider(<Backdrop />);
    expect(asFragment()).toMatchSnapshot();
  });
});
