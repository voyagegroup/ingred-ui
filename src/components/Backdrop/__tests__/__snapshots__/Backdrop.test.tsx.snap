// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop component testing Backdrop 1`] = `
<DocumentFragment>
  <div
    class="sc-gsDKAQ fFBudr fade-transition-appear fade-transition-appear-active"
  >
    <div
      class="sc-bdvvtL cQENSx"
    />
  </div>
</DocumentFragment>
`;
